### ISSUE TITLE
autocmd User FireplaceEvalPost for integration with vim-clojure-highlight

### ISSUE THREAD

**Issue by dan-santana** [tpope/vim-fireplace#232]
I'm trying to integrate vim-fireplace and [vim-clojure-highlight](https://github.com/guns/vim-clojure-highlight) on my setup. The idea is that whenever a code is evaluated, the vim-clojure-highlight's `:ClojureHighlightReferences` will be called to refresh the highlighting in a more semantic way.

The problem is that I found no way to call it on the `cp{motion}` mappings, and calling `:ClojureHighlightReferences` manually everytime to just refresh the highlighting is really, really boring (yes, a key mapping is less boring... but I would love to avoid that "F5" key for my syntax everytime I  eval my code).

There's any possibility to add a `FireplaceEvalPost` hook? So it will be able to refresh the highlighting automatically by calling `:ClojureHighlightReferences` whenever something was evaluated.

If you think that it's outside the scope of the project providing such hook, do you have any solution for this setup? My current workaround is to bind `autocmd BufWritePost *.clj silent! Require | ClojureHighlightReferences` and to write the buffer to trigger the refresh... but sometimes I just don't want to write the buffer... mainly when I made a change on the code with side effects just to test it and I forget that my [Ring](https://github.com/ring-clojure/ring) server will automatically reload it when it's rewritten.

**Comment by tpope** [comment]
Feel free to spike out an implementation. I'm not sure the appropriate place to call it but `fireplace#session_eval()` is probably good enough for testing purposes.

My primary concern is that messing with syntax highlighting during an eval could trigger a redraw, and redraws tend to clear echoed output.

**Comment by dan-santana** [comment]
Sorry for the delay, I got involved with other things and couldn't try it until now.

I just made a `execute ':ClojureHighlightReferences'` call right after [plugin/fireplace.vim#L786](https://github.com/tpope/vim-fireplace/blob/8579a41a186702e20bb25e21ebaa5a3a1b9c265a/plugin/fireplace.vim#L786), and it worked perfectly! The output wasn't overriden, so if there's somewhere to call a `FireplaceEvalPre` and `FireplaceEvalPost` autocommands, probably it's around this line.

**NOTE**: Actually the highlighting ran **a lot** faster than when writing the file, probably because I didn't `:Require` everything again... so probably there won't be any performance issues for any other user who might want to try this setup.
